### PR TITLE
[C-1452] Implement release-date field

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Implements release-date field
- Adds react-native-modal-datetime-picker

https://user-images.githubusercontent.com/8230000/201187632-cb52848a-bcfe-42e7-80e1-8f719ce94281.mov

### CODE REVIEW

File: packages/mobile/src/screens/upload-screen/fields/ReleaseDateField.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import { makeStyles } from 'app/styles'
+import { useThemeColors, useThemeVariant } from 'app/utils/theme'
+
+const isToday = (date: Date) => {
+  const today = new Date()
+  return (
+    date.getDate() === today.getDate() &&
+    date.getMonth() === today.getMonth() &&
+    date.getFullYear() === today.getFullYear()
+  )
```
Comment (amendelsohn) [inline packages/mobile/src/screens/upload-screen/fields/ReleaseDateField.tsx:24]: 😅 it really feels like there should be an easier way, but this totally works. No complaints

File: packages/mobile/src/screens/upload-screen/fields/ReleaseDateField.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  },
+  label: {
+    marginTop: spacing(1)
+  },
+  dateText: {
+    marignTop: spacing(1),
+    textTransform: 'uppercase'
+  },
+  divider: {
+    marginHorizontal: spacing(-4)
```
Comment (amendelsohn) [inline packages/mobile/src/screens/upload-screen/fields/ReleaseDateField.tsx:49]: 👀

File: packages/mobile/src/screens/upload-screen/fields/ReleaseDateField.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        onChange('release_date')(dateString)
+      }
+      handleClose()
+    },
+    [onChange, handleClose]
+  )
+
+  return (
+    <>
+      <TouchableOpacity style={styles.root} onPress={handlePress}>
```
Comment (amendelsohn) [inline packages/mobile/src/screens/upload-screen/fields/ReleaseDateField.tsx:118]: nit: `root` feels slightly misleading

maybe best fixed by moving the divider up a level?

**Review by amendelsohn — APPROVED** [review]

File: packages/mobile/src/screens/upload-screen/fields/ReleaseDateField.tsx (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/upload-screen/fields/ReleaseDateField.tsx:24]: There def is I'll use moment

File: packages/mobile/src/screens/upload-screen/fields/ReleaseDateField.tsx (hunk 3 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/upload-screen/fields/ReleaseDateField.tsx:118]: Yes good call

**Pull request merged**